spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -215,8 +231,12 @@ public void destroy() {
 	public void addTopic(String topic, int qos) {
 		this.topicLock.lock();
 		try {
-			this.mqttClient.subscribe(topic, qos).waitForCompletion(getCompletionTimeout());
-			super.addTopic(topic, qos);
+			if (mqttClient != null && mqttClient.isConnected()) {
+				this.mqttClient.subscribe(topic, qos).waitForCompletion(getCompletionTimeout());
+				super.addTopic(topic, qos);
+			} else {
+				logger.warn(String.format("An attempt to add topic : '%s' with QoS : '%s' without having mqtt client initialized or connected, the topic will not be added!", topic, qos));
```

Comment:
I don't think it is so bad to store topics (`super.addTopic(topic, qos);`) for the future use.
Something similar what we have so far in the `MqttPahoMessageDrivenChannelAdapter`.
So, please, change the logic over here to the old one and subscribe only if there is an `mqttClient`.

Reply:
Well, ok, perhaps) I just retained a warning in case `mqttClient` is not initialized